src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      const upgradeType = semverUpgradeType(
+        testVersion,
+        testConfig.latestVersion as string
+      );
+      vc.log();
+
+      assert.strictEqual(
+        upgradeType && message.indexOf(upgradeType) >= 0,
+        true,
+        "Log does not contain the correct upgradeType."
+      );
+    });
+    it("logs the message", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      vc.log();
+
+      assert.strictEqual(
+        message.length > 0,
+        true,
+        "Log does not display the message."
+      );
+    });
+    it("sets the latest version", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      assert.notStrictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "latestVersion and lastVersionLogged is the same before logging."
+      );
+
+      vc.log();
+
+      assert.strictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "lastVersionLogged was not successfully set after logging version message."
+      );
+    });
+    it("sets lastNotification", async () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      assert.notStrictEqual(
+        vc._config.lastVersionLogged,
+        testConfig.latestVersion,
+        "latestVersion and lastVersionLogged is the same before logging."
+      );
+
+      const firstNotification = vc._config.lastNotification;
+
+      vc.log();
+
+      const secondNotification = vc._config.lastNotification;
+
+      assert.strictEqual(
+        secondNotification > firstNotification,
+        true,
+        "lastVersionLogged was not successfully set after logging version message."
+      );
+    });
+    it("only logs the latestVersion one time", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+
+      vc.log();
+
+      assert.strictEqual(
+        message.length > 0,
+        true,
+        "Log did not log the first time for this version."
+      );
+      message = "";
+      vc.log();
+
+      assert.strictEqual(message, "", "logged the same version message twice.");
+    });
+
+    it("message contains the upgradeType", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      const upgradeType = semverUpgradeType(
+        vc._currentVersion,
+        vc._config.latestVersion
+      );
+
+      assert.strictEqual(
+        upgradeType && message.indexOf(upgradeType) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+
+    it("message contains the packageName", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.packageName) >= 0,
+        true,
+        "Message does not contain the upgradeType."
```

Comment:
```suggestion
        "Message does not contain the packageName."
```

Reply:
Which suggestion takes priority? These message changes or the suggestion that messages do not provide value here: https://github.com/trufflesuite/ganache/pull/3285#discussion_r1035490385